nova_vm/src/engine/bytecode/executable.rs

```diff
@@ -561,6 +561,46 @@ impl CompileEvaluation for ast::BinaryExpression<'_> {
     }
 }
 
+impl CompileEvaluation for ast::LogicalExpression<'_> {
+    fn compile(&self, ctx: &mut CompileContext) {
+        self.left.compile(ctx);
+        if is_reference(&self.left) {
+            ctx.exe.add_instruction(Instruction::GetValue);
+        }
+        // We store the left value on the stack, because we'll need to restore
+        // it later.
+        ctx.exe.add_instruction(Instruction::LoadCopy);
+
+        match self.operator {
+            oxc_syntax::operator::LogicalOperator::Or => {
+                ctx.exe.add_instruction(Instruction::LogicalNot);
+            }
+            oxc_syntax::operator::LogicalOperator::And => {}
+            oxc_syntax::operator::LogicalOperator::Coalesce => {
+                ctx.exe.add_instruction(Instruction::IsNullOrUndefined);
+            }
+        }
+        let jump_to_return_left = ctx
```

Comment:
**thought:** I wonder if it would be worthwhile to add a `JumpIfNotWithStore` kind of instruction...

Reply:
I think it might be better to have a version of `JumpIfNot` that switches on `vm.result` without taking it, since the only reason we need to duplicate the result into the stack is because the result gets lost after the jump.

Reply:
Wouldn't we also need to restore it from the stack in the `Not` and `IsNullOrUndefined` cases, to undo their work?

Reply:
Oh, that's right.

In any case, I think as long as the result is independent from the stack, there will have to be a way to duplicate the result into the stack and to load the top of the stack into the result without popping. In that sense, `LoadCopy` seems a lot more general than `JumpIfNotWithStore`.

Reply:
Yeah, I like `LoadCopy` a lot, it's a great addition/idea by you.